benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/tasks/running/CheckTrialResultTask.java

```diff
@@ -67,22 +71,23 @@ public TrialResult call() throws Exception {
 
       }
 
-      if (trialResult == TrialResult.FAILURE && retries >= maxRetries) {
-        trialResult = TrialResult.EXECUTION_FAILURE;
+      if (failureStatus == null) {
+        return TrialExecutionStatus.RE_EXECUTE_TRIAL;
+      } else {
```

Comment:
If `failureStatus = FailureStatus.EXECUTION;` here you set `TrialExecutionStatus.SUCCESS`.